src/QMCWaveFunctions/Fermion/MatrixUpdateCUDA.h

```diff
@@ -0,0 +1,314 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2020 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_MATRIX_UPDATE_CUDA_H
+#define QMCPLUSPLUS_MATRIX_UPDATE_CUDA_H
+
+#include "simd/allocator.hpp"
+#include "Platforms/PinnedAllocator.h"
+#include "OpenMP/OMPallocator.hpp"
+#include "OhmmsPETE/OhmmsVector.h"
+#include "OhmmsPETE/OhmmsMatrix.h"
+#include "QMCWaveFunctions/Fermion/DiracMatrix.h"
+#include "Platforms/OpenMP/ompBLAS.hpp"
+#include "Platforms/OpenMP/ompReduction.hpp"
+#include <cuda_runtime_api.h>
+#include "Numerics/CUDA/cuBLAS.hpp"
+#include "Platforms/CUDA/cuBLAS_inhouse.hpp"
+#include "QMCWaveFunctions/Fermion/matrix_update_helper.hpp"
+
+namespace qmcplusplus
+{
+/** implements dirac matrix update using OpenMP
```

Comment:
Should say MatrixUpdateCUDA implements a specialized type of the generic DET_ENGINE type.

Reply:
Updated comments.